Mathlib/Data/Polynomial/RingDivision.lean

```diff
@@ -239,6 +239,19 @@ theorem isUnit_iff : IsUnit p ↔ ∃ r : R, IsUnit r ∧ C r = p :=
     fun ⟨_, hr, hrp⟩ => hrp ▸ isUnit_C.2 hr⟩
 #align polynomial.is_unit_iff Polynomial.isUnit_iff
 
+theorem not_isUnit_of_degree_pos [IsDomain R] (p : R[X])
+    (hpl : 0 < p.degree) : ¬ IsUnit p := by
+  intro h
+  simp [degree_eq_zero_of_isUnit h] at hpl
+
+theorem not_isUnit_of_natDegree_pos [IsDomain R] (p : R[X])
+    (hpl : 0 < p.natDegree) : ¬ IsUnit p :=
+  not_isUnit_of_degree_pos _ (natDegree_pos_iff_degree_pos.mp hpl)
+
+lemma natDegree_mul_C_leadingCoeff_inv {K} [Field K] {p : K[X]} (hp0 : p ≠ 0) :
```

Comment:
This is also true if `p = 0`, right? And `IsDomain` is enough.

Also, it seems more reasonable to assume `a ≠ 0` (for `a : R`) and state that `(p * a).natDegree = p.natDegree` (applying this to the leading coefficient is immediate via `leadingCoeff_ne_zero`).

Reply:
> And IsDomain is enough

The problem is that we can't write `(leadingCoeff p)⁻¹` in a domain. Some parts of mathlib use `IsUnit (leadingCoeff p)`, which would be a large refactor to implement here. (I'm okay with doing that in a PR that this will depend on.)

Reply:
Ah you're right. Then I suggest to prove (in a domain) that the degree doesn't change when we multiply the polynomial by a nonzero constant, and prove this version (for fields) using that.

Anyway `p ≠ 0` seems useless.

Reply:
That result already exists: [Polynomial.(nat)degree_mul_C](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Polynomial/Degree/Lemmas.html#Polynomial.degree_mul_C). I've moved this lemma to the other `mul_leadingCoeff` lemmas and restated it in terms of that lemma.